File: src/main/java/com/bookislife/jgsk/java/_19_inherit/Button.java
package com.bookislife.jgsk.java._19_inherit;

/**
 * Created by SidneyXu on 2015/09/29.
 */
public class Button extends View implements OnClickListener {

    private String text;

    public Button(String text, int width, int height) {
        this.text = text;
        this.width = width;
        this.height = height;
    }

    @Override
    public void onClick() {
        System.out.println("Click Button " + text);
    }

    @Override
    public String getDescription() {
        return text;
    }
}

File: src/main/scala/com/bookislife/jgsk/scala/_19_inherit/Button.scala
package com.bookislife.jgsk.scala._19_inherit

/**
  * Created by Sidney on 2015/09/29.
  */
class Button(val text: String, width: Int, override val height: Int) extends View(width, height) {
  //  Overrides abstract method
  def getDescription(): String = text

  override def toString: String = getDescription()
}
